composer/rest/composer2/composer2_airflow_rest_api.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+Trigger a DAG in Cloud Composer 2 environment using the Airflow 2 stable REST API.
+"""
+
+# [START composer_2_trigger_dag]
+
+import google.auth
+from google.auth.transport.requests import AuthorizedSession
+
+AUTH_SCOPE = "https://www.googleapis.com/auth/cloud-platform"
+
+
+def make_composer2_web_server_request(url, method="GET", **kwargs):
+    """
+    Make a request to Cloud Composer 2 environment's web server.
+    Args:
+      url: The URL to fetch.
+      method: The request method to use ('GET', 'OPTIONS', 'HEAD', 'POST', 'PUT',
+        'PATCH', 'DELETE')
+      **kwargs: Any of the parameters defined for the request function:
+                https://github.com/requests/requests/blob/master/requests/api.py
+                  If no timeout is provided, it is set to 90 by default.
+    """
+
+    credentials, _ = google.auth.default(scopes=[AUTH_SCOPE])
+    authed_session = AuthorizedSession(credentials)
+
+    # Set the default timeout, if missing
+    if "timeout" not in kwargs:
+        kwargs["timeout"] = 90
+
+    return authed_session.request(method, url, **kwargs)
+
+
+def trigger_dag(web_server_url, dag_id, data):
+    """
+    Make a request to trigger a dag using the stable Airflow 2 REST API.
+    https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html
+
+    Args:
+      web_server_url: The URL of the Airflow 2 web server.
+      dag_id: The DAG ID.
+      data: Additional configuration parameters for the DAG run (json).
+    """
+
+    endpoint = f"api/v1/dags/{dag_id}/dagRuns"
+    request_url = f"{web_server_url}/{endpoint}"
+    json_data = {"conf": data}
+
+    response = make_composer2_web_server_request(
+        request_url, method="POST", json=json_data
+    )
+
+    if response.status_code == 403:
+        raise Exception(
```

Comment:
Exception is rather generic.

Normally, I'd expect a [Forbidden](https://googleapis.dev/python/google-api-core/latest/exceptions.html#google.api_core.exceptions.Forbidden) exception, but you might prefer [HTTPError](https://docs.python-requests.org/en/master/api/#requests.HTTPError) since you're using `requests`.